class-gf-razorpay.php

```diff
@@ -0,0 +1,336 @@
+            'razorpay_signature'  => rgpost('razorpay_signature'),
+        );
+
+        $success = false;
+
+        if (($entry !== null) and
+            (empty(rgpost('razorpay_payment_id')) === false) and
+            (empty(rgpost('razorpay_signature')) ===false))
+        {
+            try
+            {
+                $api->utility->verifyPaymentSignature($attributes);
+                $success = true;
+            }
+            catch (\Exception $e)
+            {
+                $success = false;
+            }
+        }
+
+        $action = array(
+            'id'             => $attributes['razorpay_payment_id'],
+            'type'           => 'complete_payment',
+            'transaction_id' => $attributes['razorpay_payment_id'],
+            'amount'         => $entry['payment_amount'],
+            'entry_id'       => $entry['id']
+        );
+
+        if ($success === false)
+        {
+            $action['type'] = 'fail_payment';
+        }
+
+        return $action;
+    }
+
+    public function post_callback($callback_action, $callback_result)
+    {
+        $entry               = GFAPI::get_entry($callback_action['entry_id']);
+        $feed                = $this->get_payment_feed($entry );
+
+        do_action('gform_razorpay_post_payment_' . $callback_action['type'],  $_POST, $entry, $feed);
+    }
+
+    public function generate_razorpay_form($entry, $customerFields, $form)
+    {
+        $feed = $this->get_payment_feed($entry, $form);
+
+        $key = $this->get_plugin_setting(self::GF_RAZORPAY_KEY);
+
+        $razorpayArgs = array(
+            'key'         => $key,
+            'name'        => $form['name'],
+            'amount'      => (int) round($entry['payment_amount'] * 100),
+            'currency'    => $entry['currency'],
+            'description' => $form['description'],
+            'prefill'     => array(
+                'name'    => $customerFields['billingInformation_firstName'],
+                'email'   => $customerFields['billingInformation_email'],
+            ),
+            'notes'       => array(
+                'gravity_forms_order_id' => $entry['id']
+            ),
+            'order_id'    => $entry['razorpay_order_id'],
+        );
+
+        $redirectUrl = '?page=gf_razorpay_callback';
+
+
+        wp_enqueue_script('razorpay_script',
+                          plugin_dir_url(__FILE__). 'script.js',
+                          array('checkout')
+        );
+
+        wp_localize_script('razorpay_script',
+                           'razorpay_script_vars',
+                           array(
+                               'data' => $razorpayArgs
+                           )
+        );
+
+        wp_register_script('checkout',
+                           'https://checkout.razorpay.com/v1/checkout.js',
+                           null,
+                           null);
+
+        wp_enqueue_script('checkout');
+
+        return $this->generate_order_form($redirectUrl);
```

Comment:
@riagarg5 $redirectUrl is being set several lines above. Please fix it. Also, you can just directly call generate_order_form('....')